#! /usr/bin/env python

PACKAGE = 'humanoid_league_image_marker'
import roslib

roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
gen.add("ball_ROI", bool_t, 0,
        "Mark region of interest ball", True)
gen.add("ball_candidates", bool_t, 0,
        "Mark ball candidates", True)
gen.add("ball", bool_t, 0,
        "Mark ball", True)

gen.add("goal_post_ROI", bool_t, 0,
        "Mark region of intereset goal posts", True)
gen.add("goal_posts", bool_t, 0,
        "Mark goal posts", True)
gen.add("goals", bool_t, 0,
        "Mark goals", True)

gen.add("obstacle_ROI", bool_t, 0,
        "Mark region of interest obstacle", True)
gen.add("obstacles", bool_t, 0,
        "Mark obstacles", True)

gen.add("line_ROI", bool_t, 0,
        "Mark region of intereset for lines", True)
gen.add("lines", bool_t, 0,
        "Mark lines", True)

exit(gen.generate(PACKAGE, "humanoid_league_image_marker", "image_marker_params"))
